src/lib/homestore/heap_chunk_selector.cpp

```diff
@@ -0,0 +1,110 @@
+#include "heap_chunk_selector.h"
+
+#include <execution>
+#include <algorithm>
+#include <sisl/logging/logging.h>
+
+namespace homeobject {
+
+//https://github.com/facebook/folly/blob/61c11d77eb9a8bdc60f673017fccfbe900125cb6/folly/AtomicHashMap.h#L42
+//AtomichashMap has a max size limit of ~18x initial size.
+//we support max of 64k chunks for now, so 64*1024/18 + 1 = 3641, we use 4000 for sufficiently initial size 
+#define DEFAULT_ATOMICHASHMAP_INITIAL_SIZE 4000
```

Comment:
we know for every HS we will go to 64K ,maybe just using 64K?

Reply:
```
static constexpr HeapChunkSelector::chunk_atomicmap_init_size() {
     return 2<<16;}
static constexpr HeapChunkSelector::pdev_atomicmap_init_size() {
     return 2<<4;}
```

Reply:
should be 1<<16(64K) and 1 <<4 ?